### PR TITLE
Check params_file arg, and use default if it has no slam_toolbox params

### PR DESCRIPTION
**Pull request by v-lopez**
## Basic Info
| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | https://github.com/ros-planning/navigation2/issues/2240|
| Primary OS tested on | Ubuntu 20.04 |
| Robotic platform tested on | simulated turtlebot, simulated TIAGo Base |

---

## Description
This is a workaround to https://github.com/ros-planning/navigation2/pull/2243 to keep foxy API stable by not changing the params_file name.

### CODE REVIEW

**Comment by v-lopez** [comment]
@SteveMacenski As discussed on https://github.com/ros-planning/navigation2/pull/2243 this is the workaround for foxy.

I've added a LogInfo when this workaround is applied, so users have some clue that their `params_file` is being ignored.

>[INFO] [launch.user]: provided params_file /home/user/nav_ws/install/nav2_bringup/share/nav2_bringup/params/nav2_params.yaml does not contain slam_toolbox parameters. Using default: /home/user/nav_ws/install/slam_toolbox/share/slam_toolbox/config/mapper_params_online_sync.yaml

File: slam_toolbox/launch/__init__.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2021 PAL Robotics S.L.
```
Comment (SteveMacenski) [inline slam_toolbox/launch/__init__.py]: This project is LGPLv2.1 due to the use of openkarto, please use the appropriate license header (or dont include like the others)

File: slam_toolbox/launch/has_node_params.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# Copyright (c) 2021 PAL Robotics S.L.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
```
Comment (SteveMacenski) [inline slam_toolbox/launch/has_node_params.py]: We have a dep on `nav2_map_server`, you can make a dep on `nav2_common` and add this to that in foxy + use it from there.

**Review by SteveMacenski — APPROVED** [review]
Clever, works for me!

**Comment by SteveMacenski** [comment]
Just licensing / not copying files around too too much (and licensing is dealt with by just moving it back to nav2_common)

**Comment by v-lopez** [comment]
Moved the file to nav2_common as suggested. See https://github.com/ros-planning/navigation2/pull/2257.

**Review by SteveMacenski — APPROVED** [review]
Did you test with the import from nav2_common?

**Comment by v-lopez** [comment]
Yes, even removed workspace and recompiled to make sure of it.

**Pull request merged**